DataConservancy/pass-ember
# Add compliance and repositories

# Overview
* Adds a functional 'compliance' section, which enumerates all applicable policies
* Allows selection of payment option for NIH form B
* Allows editing and re-calculating of policies, repositories if journal or funders are changed
* Allows adding repositories _not_ implied by policy
* Adds simple transitions to and from metadata section

# Known Issues
* Filtering metadata forms has not been implemented;  all metadata forms are shown regardless of whether they are necessary or not.  This is simply because Iran out of time.

# To test
# How to test
* Check out this PR locally 
  1.  Check out the latest `master` from pass-ember (e.g. `git checkout master` then `git pull`)
  2.  Create a branch from master to try it out in : `git checkout -b birkland-issue-19 master`
  3.  Pull the PR into it `git pull https://github.com/birkland/pass-ember.git issue-19`
  4.  Do `npm install` and then `ember serve`
* Create a new submission and start playing around
  1. Choose the _Food & Function_ journal for an example of NIH B
  2. Pick a random NIH grant,  like "Identification and Activation mechanisms..."
  3. You should see the NIH policy, with radio button options
      1. Click the top "I have made", then save and continue.  The next repos page should show no repositories
      2. Add a repository - let's say JHU-IR
      3. Go back to compliance.  Now click the other (bottom) radio button "will not make an arrangement"; save and continue.  You should see PMC in the list of repositories, along with your JHU-IR preference
   4. Now navigate all the way back to awards.  Add an NSF award by searching for 'career', you'll see a CAREER award.  Save and continue.  You should see the NSF policy in along with NIH in compliance
      * Click save and continue to go to repositories.  Now you should see three
  5. Now go WAY back to 'overview' and edit the journal.  Pick a "method A" journal like the AAPS journal.  Save and continue.
      * Go to compliance.  you should see NSF and NIH, but the NIH wiill be different.  It'll say you don't have to submit

Interested parties: @htpvu 

Resolves: #19
